fix assets_url attribute typo

smiller171 opened the pull request:
I noticed that the attribute `assets_url` was typod to be `asserts_url`. This PR is my attempt to fix that mistake without breaking anyone's existing Terraform code.

~~I'm not sure I've handled the deprecation message properly, but I'm happy to make any adjustments needed.~~

I definitely didn't do the deprecation message right and have removed it to allow the tests to pass. Any guidance on how to implement the deprecation message would be appreciated.

jcudit commented:
The CI failure looks related to a [previously merged PR](https://github.com/terraform-providers/terraform-provider-github/pull/424).

```
ERRO Running error: varcheck: analysis skipped: errors in package: [/home/travis/gopath/src/github.com/terraform-providers/terraform-provider-github/main.go:5:2: could not import github.com/terraform-providers/terraform-provider-github/github (/home/travis/gopath/src/github.com/terraform-providers/terraform-provider-github/github/config.go:11:2: could not import github.com/google/go-github/v31/github (/home/travis/gopath/src/github.com/terraform-providers/terraform-provider-github/vendor/github.com/google/go-github/v31/github/github-accessors.go:11338:2: undeclared name: dataSource))] 
```


/cc @anGie44 in case this looks familiar ☝️

smiller171 commented:
@jcudit No, that CI failure was from me copying code without understanding it to try to make a deprecation message work.

anGie44 commented on github/data_source_github_release.go:
```diff
@@ -147,7 +151,8 @@ func dataSourceGithubReleaseRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("url", release.GetURL())
 	d.Set("html_url", release.GetHTMLURL())
-	d.Set("asserts_url", release.GetAssetsURL())
+	d.Set("asserts_url", release.GetAssertsURL())
```
hi @smiller171! I'd recommend leaving this as `d.Set("asserts_url", release.GetAssetsURL())` so existing users/configs can still access this value. If we go in the direction of Deprecating the `asserts_url` attribute, within the "assets_url" block, after L87, you can add the key/value pair: `Deprecated: "Use `assets_url` instead",`. these docs provide additional examples: https://www.terraform.io/docs/extend/best-practices/deprecations.html#renaming-a-computed-attribute

smiller171 replied:
Thanks for the link. I hope to have time to fix this soon

anGie44 commented on vendor/github.com/google/go-github/v31/github/github-accessors.go:
```diff
@@ -11332,6 +11332,16 @@ func (r *RepositoryPermissionLevel) GetUser() *User {
 	return r.User
 }
 
+// GetAssertsURL returns the AssetsURL field if it's non-nil, zero value otherwise. Created to generate a deprecation warning
```
this custom method can be safely omitted here as we can reuse `GetAssetsURL()` to populate the misspelled attribute

smiller171 replied:
My intent with the extra method was to have a place to handle the deprecation message. I haven't made the time to go through the deprecation docs you linked, but hopefully I'll get to that soon

anGie44 commented on github/data_source_github_release.go:
```diff
@@ -82,6 +82,10 @@ func dataSourceGithubRelease() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"assets_url": {
```
looks great and follows the terraform docs recommendations 👌

jcudit commented:
@smiller171 are you around to take care of the conflict here? 🤔  seems like we can remove the edit to `vendor/github.com/google/go-github/v31/github/github-accessors.go` and we'll be good to merge.

smiller171 commented:
@jcudit I've rebased and reverted the change to `github-accessors`. It's been a very long time since I looked at this, and I know I never came back to implement that deprecation message.

Also it looks like a bunch of tests are failing now...

jcudit commented:
Ah, didn't realize there was more to be done here.  We can circle back on this effort another day.

nickfloyd commented:
👋 Hey Friends, this issue has been automatically marked as stale because it has no recent activity. It will be closed if no further activity occurs. Please add the `Status: Pinned` label if you feel that this issue needs to remain open/active. Thank you for your contributions and help in keeping things tidy!
